packages/cadl-ranch-specs/http/azure/client-generator-core/format/main.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Describe scenarios for various combinations of unions.
+ */
+@supportedBy("dpg")
+@scenarioService("/azure/client-generator-core/format")
+namespace _Specs_.Azure.ClientGenerator.Core.Format;
+
+model Model {
+  @format("guid")
+  guidProperty?: string;
+
+  @format("url")
+  urlProperty?: string;
+
+  @format("uuid")
+  uuidProperty?: string;
+
+  @format("armId")
+  armIdProperty?: string;
+
+  @format("ipAddress")
+  ipAddressProperty?: string;
+
+  @format("azureLocation")
+  azureLocation?: string;
+
+  @format("etag")
+  etagProperty?: string;
```

Comment:
How many valid stuff for `@format`? I can't find any doc describes it.

Reply:
`url` should be a scalar. https://microsoft.github.io/typespec/language-basics/built-in-types#string-types

`@format` seems just the "format" in swagger, which can be anything.

For Java, these are just all `String`. So no opinion from me...

Reply:
> `url` should be a scalar. [microsoft.github.io/typespec/language-basics/built-in-types#string-types](https://microsoft.github.io/typespec/language-basics/built-in-types#string-types)
> 
> `@format` seems just the "format" in swagger, which can be anything.
> 
> For Java, these are just all `String`. So no opinion from me...

it is kind of required for dotnet MPG, therefore I introduced these cases